pkg/querypb/query_response_extra_test.go

```diff
@@ -0,0 +1,69 @@
+package querypb
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/grafana/regexp"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAllPrometheusStatusValues(t *testing.T) {
+	prometheusStatusStrings := extractPrometheusStatusStrings(t)
+
+	for _, prometheusStatusString := range prometheusStatusStrings {
+		status, err := StatusFromPrometheusString(prometheusStatusString)
+		require.NoError(t, err)
+
+		actualStatusString, err := status.ToPrometheusString()
+		require.NoError(t, err)
+		require.Equal(t, prometheusStatusString, actualStatusString)
+	}
+}
+
+func TestAllPrometheusErrorTypeValues(t *testing.T) {
+	prometheusErrorTypeStrings := extractPrometheusErrorTypeStrings(t)
+
+	for _, prometheusErrorTypeString := range prometheusErrorTypeStrings {
+		errorType, err := ErrorTypeFromPrometheusString(prometheusErrorTypeString)
+		require.NoError(t, err)
+
+		actualErrorTypeString, err := errorType.ToPrometheusString()
+		require.NoError(t, err)
+		require.Equal(t, prometheusErrorTypeString, actualErrorTypeString)
+	}
+}
+
+// HACK: this is a very fragile way of checking if there have been any additional status values added to Prometheus
```

Comment:
I don't love what I've done here, very open to feedback and alternative suggestions.